kd/deployment.yaml

```diff
@@ -0,0 +1,255 @@
+          - name: DB_NAME
+            valueFrom:
+              secretKeyRef:
+                name: {{.KUBE_NAMESPACE}}-rds
+                key: db_name
+          - name: DB_SCHEMA_NAME
+            valueFrom:
+              secretKeyRef:
+                name: hocs-workflow
+                key: rds_schema
+          - name: DB_USERNAME
+            valueFrom:
+              secretKeyRef:
+                name: hocs-workflow
+                key: rds_user
+          - name: DB_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: hocs-workflow
+                key: rds_password
+        resources:
+          limits:
+            cpu: 900m
+            memory: 512Mi
+          requests:
+            cpu: 200m
+            memory: 80Mi
+        ports:
+          - name: http
+            containerPort: 8080
+        livenessProbe:
+          httpGet:
+            path: /actuator/health
+            port: http
+            httpHeaders:
+              - name: X-probe
+                value: kubelet
+          initialDelaySeconds: 45
+          periodSeconds: 20
+          timeoutSeconds: 10
+        readinessProbe:
+          httpGet:
+            path: /actuator/health
+            port: http
+            httpHeaders:
+              - name: X-probe
+                value: kubelet
+          initialDelaySeconds: 45
+          periodSeconds: 20
+          timeoutSeconds: 10
+        volumeMounts:
+          - mountPath: /certs
+            name: certs
+            readOnly: true
+          - mountPath: /etc/ssl/certs
+            name: bundle
+            readOnly: true
+      
+      - name: keycloak-proxy-web
+        image: quay.io/keycloak/keycloak-gatekeeper:8.0.2
+        imagePullPolicy: Always
+        securityContext:
+          runAsNonRoot: true
+        resources:
+          limits:
+            memory: 32Mi
+            cpu: 400m
+          requests:
+            memory: 8Mi
+            cpu: 100m
+        env:
+          - name: ENCRYPTION_KEY
+            valueFrom:
+              secretKeyRef:
+                name: hocs-frontend
+                key: encryption_key
+        args:
+          - --config=/etc/secrets/data.yml
+          - --discovery-url=https://sso-dev.notprod.homeoffice.gov.uk/auth/realms/hocs-notprod
+          - --openid-provider-proxy=http://hocs-outbound-proxy.{{.KUBE_NAMESPACE}}.svc.cluster.local:31290
+          - --listen=127.0.0.1:8081
+          - --enable-logging=true
+          - --enable-json-logging=true
+          - --upstream-url=http://127.0.0.1:8080 # app
+          - --upstream-response-header-timeout=60s
+          - --upstream-expect-continue-timeout=60s
+          - --upstream-keepalive-timeout=60s
+          - --server-read-timeout=60s
+          - --server-write-timeout=60s
+          - --no-redirects=false
+          - --resources=uri=/health|white-listed=true
+          - --resources=uri=/public/*|white-listed=true
+          - --resources=uri=/*
```

Comment:
Consider hiding this behind a Keycloak role -- otherwise everyone with any notprod account will be able to access this. Unless that's what we want?